src/main/java/tictactoe/core/Game.java

```diff
@@ -6,64 +6,56 @@
 public class Game {
 
     private Board board;
-    private final Players players;
-    private final UI ui;
+    private Players players;
+    private Mediator mediator;
 
-    public Game(int boardSize, UI ui, Players players) {
-        this.board = new Board(boardSize);
-        this.ui = ui;
-        this.players = players;
+    public Game(Mediator mediator) {
+        this.mediator = mediator;
```

Comment:
This change creates a temporal coupling - so opens it up to runtime errors.

Reply:
hmm yea I was unsure how I could make that safer, i wanted the `receivePlayers` and `receiveBoardSize` methods separate as they'd be getting possibly asynchronous values from the UI. Is it worth doing null checks on the players and board or should I be doing something else?